src/main/java/org/gooru/insights/services/ItemServiceImpl.java

```diff
@@ -465,31 +466,32 @@ public void generateReport(String traceId, String data, String sessionToken, Map
 
 		ColumnList<String> reportConfig = getBaseConnectionService().getColumnListFromCache(CassandraRowKeys.EXPORT_REPORT_CONFIG.CassandraRowKey());
 		int maxLimit = reportConfig.getIntegerValue(APIConstants.MAXIMUM_ROW_LIMIT, 0);
-		int totalRows = 0;
-		String fileName = UUID.randomUUID().toString();
-		final String absoluteFilePath = getBaseConnectionService().getRealRepoPath().concat(fileName).concat(APIConstants.DOT).concat(APIConstants.CSV_EXTENSION);
+		int requestedRowLimit = 0;
+		String fileName = UUID.randomUUID().toString().substring(0, 8).concat(APIConstants.FORWARD_SLASH).concat(APIConstants.EXPORT_FILE_NAME).concat(APIConstants.DOT).concat(APIConstants.CSV_EXTENSION);
```

Comment:
In random number generator we could get number with in a range. And it will generate duplicate number in some scenario.